### PR TITLE
Add a working date range picker

### PR DESCRIPTION
**Pull request by nachtm**
Now our date picker picks and filters a whole range of dates! It looks like this:
image

Selecting a start date:
image
Selecting an end date when the start date has already been chosen:
image
Selecting a start date when the end date has already been chosen: 
image
Selecting a start date when the start date has already been chosen:
image

### CODE REVIEW

File: src/services/FilterService.js [hunk 1]
```diff
@@ -75,7 +78,8 @@ export const getInitialFilter = (allNeighborhoods) => {
     const initialFilter = {
```
Comment (sayas01) [inline src/services/FilterService.js]: not part of this PR but can we simply return this instead of declaring this variable?

File: src/components/FilterDrawer.js [hunk 2]
```diff
@@ -186,11 +200,6 @@ class FilterDrawer extends React.Component {
                         </FormControl>
                     )}
                 </div>
-                <Button variant="contained"
```
Comment (sayas01) [inline src/components/FilterDrawer.js]: 👍

File: src/services/FilterService.js [hunk 3]
```diff
@@ -33,14 +35,15 @@ const DATE_BOUNDS = [
...
+ * OR (target is before endDate AND target is after startDate)
+ * @param target
+ * @param startDate
+ * @param endDate
  */
-const daysAreEqual = (one, two) => {
-    return one.getUTCFullYear() === two.getUTCFullYear() 
-        && one.getUTCMonth() === two.getUTCMonth()
-        && one.getUTCDate() === two.getUTCDate();
+const insideDateBounds = (target, startDate, endDate) => {
```
Comment (sayas01) [inline src/services/FilterService.js]: 👍

File: src/services/FilterService.js (hunk 1 above)
Comment (nachtm) [reply src/services/FilterService.js]: I think so? I've added it to the quip doc to look at later.
